service/controller/app/v1/appcatalog/appcatalog.go

```diff
@@ -0,0 +1,76 @@
+package appcatalog
```

Comment:
The appcatalog package is used to fetch the CR because its needed by chart, configmap and secret resources. It will be added to the context so we only call the K8s API once.

Reply:
That is nice, but when the catalog is a CR, why don't we watch the CR?

Reply:
I'm working on that another `index` resource which listening to appCatalog changes, maybe it would answer for yours?

Reply:
Yes like @tomahawk28 says the operator has 2 controllers with different purposes.

The appCatalog is cluster scoped and acts as a "reference CR" storing catalog level configuration. So to calculate the desired state we need both CRs.

Reply:
I see. Thank you.